node/impl/full.go

```diff
@@ -117,4 +123,84 @@ func (n *FullNodeAPI) NodeStatus(ctx context.Context, inclChainStatus bool) (sta
 	return status, nil
 }
 
+// This method is exclusively used for Non Interactive Authentication based on SP Worker Address
+func (n *FullNodeAPI) FilIdSp(ctx context.Context, maddr address.Address) (string, error) {
+	filAuthHdr := "FIL-SPID-V0"
+
+	head, err := n.ChainHead(ctx)
+	if err != nil {
+		return "", xerrors.Errorf("failed to get chain head: %w", err)
+	}
+
+	beacon, err := n.StateGetBeaconEntry(ctx, head.Height())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get DRAND beacon: %w", err)
+	}
+
+	finality := head.Height() - miner.ChainFinality
+	ftipset, err := n.ChainGetTipSetByHeight(ctx, finality, head.Key())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get finality tipset: %w", err)
+	}
+
+	msg := make([]byte, 0, 99)
+
+	// Append prefix to avoid making a viable message
+	// FIXME: this should be replaced by SPIDV1 combined with proper domain-separation when it arrives
+	msg = append(msg, []byte("   ")...)
+	// Append beacon
+	msg = append(msg, beacon.Data...)
+
+	// Get miner info for last finality
+	minfo, err := n.StateMinerInfo(ctx, maddr, ftipset.Key())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get miner info: %w", err)
+	}
+
+	sig, err := n.WalletSign(ctx, minfo.Worker, msg)
+	if err != nil {
+		return "", xerrors.Errorf("failed to sign message with worker address key: %w", err)
+	}
+
+	return fmt.Sprintf("%s %d;%s;%s", filAuthHdr, head.Height(), maddr, base64.StdEncoding.EncodeToString(sig.Data)), nil
+}
+
+// This method is exclusively used for Non Interactive Authentication based on provided Wallet Address
+func (n *FullNodeAPI) FilIdAddr(ctx context.Context, addr address.Address) (string, error) {
+	filAuthHdr := "FIL-ADDRID-V0"
+
+	head, err := n.ChainHead(ctx)
+	if err != nil {
+		return "", xerrors.Errorf("failed to get chain head: %w", err)
+	}
+
+	if addr.Protocol() == address.ID {
+		resolvedAddr, err := n.StateAccountKey(ctx, addr, head.Key())
+		if err != nil {
+			return "", xerrors.Errorf("could not find public key address: %w", err)
+		}
+		addr = resolvedAddr
+	}
+
+	beacon, err := n.StateGetBeaconEntry(ctx, head.Height())
+	if err != nil {
+		return "", xerrors.Errorf("failed to get DRAND beacon: %w", err)
+	}
+
+	msg := make([]byte, 0, 99)
+
+	// Append prefix to avoid making a viable message
+	// FIXME: this should be replaced by ADDRIDV1 combined with proper domain-separation when it arrives
+	msg = append(msg, []byte("   ")...)
+	// Append beacon
+	msg = append(msg, beacon.Data...)
+
+	sig, err := n.WalletSign(ctx, addr, msg)
+	if err != nil {
+		return "", xerrors.Errorf("failed to sign message with worker address key: %w", err)
```

Comment:
@LexLuthr we missed an incorrect copy/paste here:

```suggestion
		return "", xerrors.Errorf("failed to sign message with wallet address key: %w", err)
```